exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/MetricAdapter.java

```diff
@@ -224,5 +246,47 @@ private static int estimateNumSamples(int numPoints, MetricDataType type) {
     return Collections.emptyList();
   }
 
+  private static Sample createSample(
+      String name,
+      List<String> labelNames,
+      List<String> labelValues,
+      double value,
+      Optional<Exemplar> exemplar) {
+    return exemplar
+        .map(e -> new Sample(name, labelNames, labelValues, value, toPrometheusExemplar(e)))
+        .orElseGet(() -> new Sample(name, labelNames, labelValues, value));
+  }
+
+  private static io.prometheus.client.exemplars.Exemplar toPrometheusExemplar(Exemplar exemplar) {
+    // TODO: Keep filtered attributes for prometheus?
+    if (exemplar.getSpanId() != null && exemplar.getTraceId() != null) {
+      return new io.prometheus.client.exemplars.Exemplar(
+          valueOf(exemplar),
+          exemplar.getRecordTimeNanos(),
+          "traceid",
+          exemplar.getTraceId(),
+          "spanid",
+          exemplar.getSpanId());
+    }
+    return new io.prometheus.client.exemplars.Exemplar(valueOf(exemplar));
+  }
+
+  private static Optional<Exemplar> filterExemplars(
```

Comment:
what do you think about moving this directly below where it's called? took me a bit to track it down while reading here.

Reply:
Done